### PR TITLE
Provide dynamic facts in Reclass hierarchy

### PR DESCRIPTION
**Pull request by simu**
Implement a new property `dynamic_facts` on the `Cluster` class and use this property to populate a new Reclass parameter
`parameters.dynamic_facts`.

This is in preparation to allow components to consume the cluster version as reported by Steward, cf. https://github.com/projectsyn/steward/issues/63

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update tests.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues.

### CODE REVIEW

**Review by corvus-ch — APPROVED** [review]

File: tests/test_cluster.py [hunk 1]
```diff
@@ -158,3 +158,15 @@ def test_facts(data):
...
     cluster = Cluster(data["cluster"], data["tenant"])
     assert facts == cluster.facts
+
+
+def test_dynamic_facts(data):
+    cluster = Cluster(data["cluster"], data["tenant"])
+    assert {} == cluster.dynamic_facts
+
+    dynamic_facts = {
+        "kubernetes_version": {"major": "1", "minor": "21", "gitVersion": "v1.21.3"}
```
[the remark this answers is not in the document]
Comment (simu) [reply tests/test_cluster.py:168]: We don't touch/inspect the string contained in `gitVersion` at any point, so it makes no difference at the moment.

File: docs/modules/ROOT/pages/reference/parameters.adoc [hunk 2]
```diff
@@ -35,3 +35,10 @@ The cloud region on which the cluster is installed.
 Mandatory only for clouds which have multiple regions.
 `distribution`::
 The Kubernetes distribution of the cluster.
+
+=== `dynamic_facts`
+
+The cluster's dynamic facts as reported from Steward on the cluster.
+
+// TODO: make k8s_version mandatory?
```
[the remark this answers is not in the document]
Comment (simu) [reply docs/modules/ROOT/pages/reference/parameters.adoc:43]: I wonder if we actually want to make the version mandatory, which is why it's currently in the PR as a question. WDYT?

[the remark this answers is not in the document]
Comment (tobru) [reply docs/modules/ROOT/pages/reference/parameters.adoc:43]: No, I wouldn't make it mandatory.

[the remark this answers is not in the document]
Comment (simu) [reply docs/modules/ROOT/pages/reference/parameters.adoc]: removed note in latest version

File: tests/test_cluster.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (corvus-ch) [reply tests/test_cluster.py:168]: I know. My thought was to have a safety net should we ever for whatever reason change this. A test would reveal that a such a change might be a brake in API.

[the remark this answers is not in the document]
Comment (simu) [reply tests/test_cluster.py]: Added a bunch of tests for different potential uses of dynamic facts

**Review by corvus-ch — APPROVED** [review]

**Review by glrf — APPROVED** [review]
LGTM

I ran it locally against Lieutenant with the dynamic fact integration and it seems to work fine.

One thing we need to coordinate: I currently call the kubernetes version fact `kubernetesVersion`. The examples here use `kubernetes_version`. I propose I will switch to `kubernetes_version` as this looks more consistent for components.

**Comment by simu** [comment]
> One thing we need to coordinate: I currently call the kubernetes version fact `kubernetesVersion`. The examples here use `kubernetes_version`. I propose I will switch to `kubernetes_version` as this looks more consistent for components.

I just used something to be able to write unit tests. We never fully agreed on whether component parameters should be snake case or camel case, so it's a mix anyway.

**Pull request merged**